ui/src/app/common/services/orb.service.ts

```diff
@@ -157,55 +137,80 @@ export class OrbService implements OnDestroy {
           .filter(this.onlyUnique),
       ),
     );
-
-    const mapTags = (list: AgentGroup[] & Sink[]) => {
-      return list
-        .map((item) =>
-          Object.entries(item.tags).map((entry) => `${entry[0]}: ${entry[1]}`),
-        )
-        .reduce((acc, val) => acc.concat(val), [])
-        .filter(this.onlyUnique);
-    };
-
-    this.groupsTags$ = this.groups$.pipe(map((groups) => mapTags(groups)));
-
-    this.sinksTags$ = this.sinks$.pipe(map((sinks) => mapTags(sinks)));
-  }
-
-  ngOnDestroy() {
-    this.killPolling.next();
-  }
-
-  getAgentListView() {
-    return this.agents$;
-  }
-
-  getAgentsTags() {
-    return this.agentsTags$;
   }
 
   getGroupsTags() {
-    return this.groupsTags$;
+    return this.observe(this.group.getAllAgentGroups()).pipe(
+      map((groups) => this.mapTags(groups)),
+    );
   }
 
   getGroupListView() {
-    return this.groups$;
+    return this.observe(this.group.getAllAgentGroups());
   }
 
   getPolicyListView() {
-    return this.policies$;
+    return this.observe(this.policy.getAllAgentPolicies());
+  }
+
+  getPolicyFullView(id: string) {
+    // retrieve policy
+    return this.policy.getAgentPolicyById(id).pipe(
+      mergeMap((policy) =>
+        // need a way to get a dataset linked to a policy without having to filter it out
+        this.dataset.getAllDatasets().pipe(
```

Comment:
@lpegoraro would it be feasible to add a filter by 'policy_id' in the dataset list endpoint?